src/Compilers/CSharp/Portable/Binder/Binder_Invocation.cs

```diff
@@ -1335,7 +1336,7 @@ private BoundExpression GetDefaultParameterSpecialNoConversion(SyntaxNode syntax
                 argsToParamsBuilder.Free();
             }
 
-            BoundExpression bindDefaultArgument(SyntaxNode syntax, ParameterSymbol parameter, Symbol containingMember, bool enableCallerInfo, BindingDiagnosticBag diagnostics)
+            BoundExpression bindDefaultArgument(SyntaxNode syntax, ParameterSymbol parameter, Symbol containingMember, bool enableCallerInfo, BindingDiagnosticBag diagnostics, ImmutableArray<BoundExpression> arguments, ImmutableArray<int> argsToParamsOpt)
```

Comment:
>ImmutableArray<BoundExpression> arguments [](start = 176, length = 41)

This doesn't feel correct. What if the expression that is being looked for is itself a parameter with a default value, located after the current parameter? I think LDM needs to answer what we should do for that case, particularly something like this:

```cs
void M([CallerArgumentExpression("param2")] string param1 = "", [CallerArgumentExpression("param1")] string param2 = "") { ... }
```

Reply:
Currently the following test

```csharp
            string source = @"
using System;
using System.Runtime.CompilerServices;

class Program
{
    public static void Main()
    {
        M();
        M(""param1_value"");
        M(param1: ""param1_value"");
        M(param2: ""param2_value"");
        M(param1: ""param1_value"", param2: ""param2_value"");
        M(param2: ""param2_value"", param1: ""param1_value"");
    }

    static void M([CallerArgumentExpression(""param2"")] string param1 = ""param1_default"", [CallerArgumentExpression(""param1"")] string param2 = ""param2_default"")
    {
        Console.WriteLine($""param1: {param1}, param2: {param2}"");
    }
}
";
```

produces the following:

```
    param1: param1_default, param2: param2_default
    param1: param1_value, param2: "param1_value"
    param1: param1_value, param2: "param1_value"
    param1: "param2_value", param2: param2_value
    param1: param1_value, param2: param2_value
    param1: param1_value, param2: param2_value
```

The current output makes sense, unless otherwise is decided in LDM.

Reply:
I haven't had confirmed `ToString` behavior for a multiline expression or with trivia. Make sure to cover those.
```cs
Assert(// comment
           expr /* comment */
              .Method());

```
I think it just ignores all that.

Reply:
@alrz Yeah I think they will be ignored. Will add a test to confirm.

(Thanks for your PR (#23861), it helped a lot ❤️ )

Reply:
Interesting case @alrz. The trivia *around* the expression is ignored, but not *inside* the expression. This might not be the expected behavior. But will wait confirmation from @333fred or someone from the compiler team.

Reply:
The trivia inside and around is a good question for LDM. Let's add a prototype comment.